staging/src/k8s.io/client-go/tools/events/doc.go

```diff
@@ -0,0 +1,19 @@
+/*
```

Comment:
While the changes in types.go require API review, these ones are obviously an improvements.
If you could split those into a separate PR, we could merge it now.

Reply:
Sure thing. Does it matter which PR is which?  Should the new PR be just the types.go changes, and this PR be the other changes; or vice-versa?

Reply:
Doesn't matter.

Reply:
OK. I've moved the non-`types.go` stuff to https://github.com/kubernetes/kubernetes/pull/85134